src/wms/src/main/java/org/geoserver/wms/capabilities/CapabilityUtil.java

```diff
@@ -43,36 +48,68 @@ public static NumberRange<Double> searchMinMaxScaleDenominator(Set<StyleInfo> st
             throws IOException {
         // searches the maximum and minimum denominator in the style's rules that are contained in
         // the style set.
-        double minScaleDenominator = Double.POSITIVE_INFINITY;
-        double maxScaleDenominator = Double.NEGATIVE_INFINITY;
+        MinMaxDenominator minMaxDenominator = new MinMaxDenominator();
+        minMaxDenominator.setMinScaleDenominator(Double.POSITIVE_INFINITY);
+        minMaxDenominator.setMaxScaleDenominator(Double.NEGATIVE_INFINITY);
```

Comment:
Sounds like the state you'd want right out of the constructor, without having to use setters from outside.

Reply:
Done